src/fastmeteo/core/grid.py

```diff
@@ -0,0 +1,114 @@
+from abc import abstractmethod
+from typing import Any
+
+import pandas as pd
+import xarray as xr
+
+
+class Grid:
+    remote_dataset: xr.Dataset
+    local_store: str
+    features: list[str]
+
+    @abstractmethod
+    def select_remote(self, hour: pd.DatetimeIndex) -> xr.Dataset: ...
+
+    @abstractmethod
+    def coords(self, flight: pd.DataFrame) -> dict[str, Any]: ...
+
+    def get_local(self, start: str | pd.DatetimeIndex) -> xr.Dataset:
+        start = pd.to_datetime(start)
+        try:
+            local_dataset = xr.open_zarr(self.local_store, consolidated=True)
+        except KeyError:
+            print(f"init local zarr from remote, hour: {start.floor('1h')}")
+            selected = self.select_remote(start.floor("1h").to_datetime64())
+            selected.to_zarr(self.local_store, mode="w", consolidated=True)
+            local_dataset = xr.open_zarr(self.local_store, consolidated=True)
+
+        return local_dataset  # type: ignore
+
+    def sync_local(
+        self, start: str | pd.DatetimeIndex, stop: str | pd.DatetimeIndex
+    ) -> xr.Dataset:
+        start = pd.to_datetime(start)
+        stop = pd.to_datetime(stop)
+
+        local_dataset = self.get_local(start)
+
+        # ensure existing and requested features are matching
+        missing_features = [
+            feature
+            for feature in self.features
+            if feature not in local_dataset.data_vars
+        ]
+        if missing_features:
+            raise RuntimeError(
+                "Requested features not in local zarr, create a new folder for this."
+            )
+
+        # ensure the data is available locally
+        for hour_dt in pd.date_range(start.floor("1h"), stop.ceil("1h"), freq="1h"):
```

Comment:
We actually have an issue here, when we switch from ERA5 to Arpege data (predictions).

Data from Arpege comes as a file every n hours (typically 6) with prediction to a given horizon (say every hour for 12 hours, but the real numbers should be documented). This means that for a given timestamp in history or short future, there maybe several versions of predictions, and you may want one of those versions (typically the latest).

Reply:
To make things simple, maybe just use the latest one.